apps/system/js/pin_page_system_dialog.js

```diff
@@ -166,6 +167,17 @@
     siteBadge.icon = data.icon;
     this.siteName.textContent = data.name;
 
+    Service.request('Places:isPinned', data.url)
+      .then((isPinned) => {
+        if (isPinned) {
```

Comment:
Yes I agree, but I just took the same syntax than pinning site just a few lines below. So leaving as is for now.